lib/concurrent-ruby/concurrent/utility/processor_counter.rb

```diff
@@ -107,4 +143,20 @@ def self.processor_count
   def self.physical_processor_count
     processor_counter.physical_processor_count
   end
+
+  # Number of processors usable for process scheduling.
+  # Returns `nil` if there is no quota, or a `Float` if the
+  # process is inside a cgroup with a dedicated CPU quota (typically Docker).
+  #
+  # For performance reasons the calculated value will be memoized on the first
+  # call.
+  #
+  # @return [nil, Float] number of usable processors seen by the OS or Java runtime
+  def self.available_processor_count
+    processor_counter.available_processor_count
```

Comment:
It feels weird to return a Float for a method named `*_count`.
Maybe we should round down instead?
And if people want a different rounding they can use `cpu_quota` directly

Reply:
I'd rather not round because it's too arbitrary. For very small local service it's not uncommon to give a very small fraction of CPU, e.g `0.5`.

So IMO it's up to the caller to decide whether they want to use a "partial" core or not. e.g. if the quota is `2.5`, should you start 2 or 3 processes to maximize utilization? Well it depends, how CPU intensive are these processes? Hence I prefer not to truncate the data.

> And if people want a different rounding they can use cpu_quota directly

That's another option, but in the context of very portable code, `cpu_quota` is likely to return `nil`, so you need to fallback on the other method. This one was really meant as some short of handy shortcut. 

But up to you.

Reply:
OK, I think I agree with the current PR & semantics then.